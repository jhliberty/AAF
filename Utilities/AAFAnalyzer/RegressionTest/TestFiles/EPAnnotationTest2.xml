<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE aaf-file SYSTEM "EPDerivationChain.dtd">
<aaf-file>
<!-- This test is the same as EPAnnotationTest, but, does not have any
         dictionary entries so it should result in extra failures -->
<!-- Test UserComments property -->
 <top-level name="User Comments in Mob Slot Segment">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <source-clip length="0" track-type="none">
    <comment value="Should Pass" name="Test Case 1"/>
    <lower-level>
     <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
      <eoc length="0" track-type="none"/>
     </timeline-mob-slot>
    </lower-level>
   </source-clip>
  </timeline-mob-slot>
 </top-level>
 <top-level name="User Comments in CommentMarker">
  <event-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <sequence length="0" track-type="none">
    <comment-marker track-type="none" length="1">
     <comment value="Should Pass" name="Test Case 2"/>
    </comment-marker>
   </sequence>
  </event-mob-slot>
 </top-level>
<!-- NOTE: Currently this test case is not causing an error because the
               comment is not in the dictionary.  This is occuring because
               the traversal ends when a UserComment is detected in the second
               sequence.  If the traversal is made to continue, this test case
               will fail -->
 <top-level name="Illegal User Comments">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <sequence length="0" track-type="none">
    <sequence track-type="none"  length="1">
     <comment value="Should Fail" name="Test Case 3"/>
     <eoc track-type="none" length="1"/>
    </sequence>
   </sequence>
  </timeline-mob-slot>
 </top-level>
<!-- Test CommentMarkes in essence/non-essence tracks -->
<!-- Note: Case 2 above tests CommentMarkers in non-essence tracks -->
 <top-level name="CommentMarker in Essence Track">
  <event-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <sequence length="0" track-type="audio">
    <comment-marker track-type="audio" length="1"/>
   </sequence>
  </event-mob-slot>
 </top-level>
 <top-level name="DescriptiveMarker in Essence Track">
  <event-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <sequence length="0" track-type="audio">
    <descriptive-marker track-type="audio" length="1"/>
   </sequence>
  </event-mob-slot>
 </top-level>
<!-- NEED TO TEST A DESCRIPTIVE MARKER IN AN ESSENCE TRACK -->
<!-- Test CommentMarkes in Mob Slots (and nested in mob slots) -->
<!-- Note: "CommentMarker in Essence Track" above test tests nested in an event mob slot -->
 <top-level name="CommentMarker in Event Mob Slot">
  <event-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <comment-marker length="0" track-type="none"/>
  </event-mob-slot>
 </top-level>
 <top-level name="CommentMarker nested in Timeline Mob Slot">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <sequence length="0" track-type="none">
    <comment-marker length="0" track-type="none"/>
   </sequence>
  </timeline-mob-slot>
 </top-level>
 <top-level name="CommentMarker in Timeline Mob Slot">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <comment-marker length="0" track-type="none"/>
  </timeline-mob-slot>
 </top-level>
 <top-level name="CommentMarker nested in Static Mob Slot">
  <static-mob-slot>
   <sequence length="0" track-type="none">
    <comment-marker length="0" track-type="none"/>
   </sequence>
  </static-mob-slot>
 </top-level>
 <top-level name="CommentMarker in Static Mob Slot">
  <static-mob-slot>
   <comment-marker length="0" track-type="none"/>
  </static-mob-slot>
 </top-level>
<!-- Note: "DescriptiveMarker in Essence Track" above test tests nested in an event mob slot -->
 <top-level name="DescriptiveMarker in Event Mob Slot">
  <event-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <descriptive-marker length="0" track-type="none"/>
  </event-mob-slot>
 </top-level>
 <top-level name="DescriptiveMarker nested in Timeline Mob Slot">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <sequence length="0" track-type="none">
    <descriptive-marker track-type="none" length="1"/>
   </sequence>
  </timeline-mob-slot>
 </top-level>
 <top-level name="DescriptiveMarker in Timeline Mob Slot">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <descriptive-marker length="0" track-type="none"/>
  </timeline-mob-slot>
 </top-level>
 <top-level name="DescriptiveMarker nested in Static Mob Slot">
  <static-mob-slot>
   <sequence length="0" track-type="none">
    <descriptive-marker track-type="none" length="1"/>
   </sequence>
  </static-mob-slot>
 </top-level>
 <top-level name="DescriptiveMarker in Static Mob Slot">
  <static-mob-slot>
   <descriptive-marker length="0" track-type="none"/>
  </static-mob-slot>
 </top-level>
<!-- Test a registered KLV Data -->
 <top-level name="Registered KLV Data">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <sequence length="0" track-type="audio">
    <klv-data value="Should Pass" key-name="KLV Test"/>
    <eoc track-type="audio" length="1"/>
   </sequence>
  </timeline-mob-slot>
 </top-level>
</aaf-file>
