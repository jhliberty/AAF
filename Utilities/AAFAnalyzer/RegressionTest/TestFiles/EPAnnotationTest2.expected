        Name:   Edit Protocol Annotation Test
        Desc:   Verify that all annotations in an AAF file are legal.
        Result: Failed
        Reason: 
        Covered Requirements: None
        Noted Requirements: None
        Passing Requirements: None
        Info Requirements: None
        Warning Requirements: None
        Failing Requirements: REQ_EP_147; REQ_EP_150; REQ_EP_151; REQ_EP_152

            Name:   Comment Marker Placement
            Desc:   CommentMarkers shall be placed only in EventMobSlots.  DescriptiveMarkers may be placed in any kind of MobSlot (e.g. TimelineMobSlot, EventMobSlot, StaticMobSlot).
            Node:   13 (CommentMarker)
            Result: Passed
            Reason: Mob slot with ID = 1 of Top-Level Composition "User Comments in CommentMarker" is an Event Mob Slot that contains a CommentMarker.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: REQ_EP_150
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: None

            Name:   Use of Component::UserComments
            Desc:   Component::UserComments shall only be used on: the Segment of a MobSlot or a CommentMarker (or its subclasses).
            Node:   18 (Sequence)
            Result: Failed
            Reason: Mob slot with ID = 1 of Top-Level Composition "Illegal User Comments" contains a Component that illegally uses the Component::UserComments property.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: None
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: REQ_EP_147

            Name:   Comment Marker Placement
            Desc:   CommentMarkers shall be placed only in EventMobSlots.  DescriptiveMarkers may be placed in any kind of MobSlot (e.g. TimelineMobSlot, EventMobSlot, StaticMobSlot).
            Node:   24 (CommentMarker)
            Result: Passed
            Reason: Mob slot with ID = 1 of Top-Level Composition "CommentMarker in Essence Track" is an Event Mob Slot that contains a CommentMarker.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: REQ_EP_150
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: None

            Name:   Comment Marker Placement
            Desc:   CommentMarkers shall be placed only in EventMobSlots.  DescriptiveMarkers may be placed in any kind of MobSlot (e.g. TimelineMobSlot, EventMobSlot, StaticMobSlot).
            Node:   31 (CommentMarker)
            Result: Passed
            Reason: Mob slot with ID = 1 of Top-Level Composition "CommentMarker in Event Mob Slot" is an Event Mob Slot that contains a CommentMarker.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: REQ_EP_150
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: None

            Name:   Comment Marker Placement
            Desc:   CommentMarkers shall be placed only in EventMobSlots.  DescriptiveMarkers may be placed in any kind of MobSlot (e.g. TimelineMobSlot, EventMobSlot, StaticMobSlot).
            Node:   35 (CommentMarker)
            Result: Failed
            Reason: Mob slot with ID = 1 of Top-Level Composition "CommentMarker nested in Timeline Mob Slot" is not an Event Mob Slot but contains a CommentMarker.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: None
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: REQ_EP_150

            Name:   Comment Marker Placement
            Desc:   CommentMarkers shall be placed only in EventMobSlots.  DescriptiveMarkers may be placed in any kind of MobSlot (e.g. TimelineMobSlot, EventMobSlot, StaticMobSlot).
            Node:   38 (CommentMarker)
            Result: Failed
            Reason: Mob slot with ID = 1 of Top-Level Composition "CommentMarker in Timeline Mob Slot" is not an Event Mob Slot but contains a CommentMarker.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: None
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: REQ_EP_150

            Name:   Comment Marker Placement
            Desc:   CommentMarkers shall be placed only in EventMobSlots.  DescriptiveMarkers may be placed in any kind of MobSlot (e.g. TimelineMobSlot, EventMobSlot, StaticMobSlot).
            Node:   42 (CommentMarker)
            Result: Failed
            Reason: Mob slot with ID = 1 of Top-Level Composition "CommentMarker nested in Static Mob Slot" is not an Event Mob Slot but contains a CommentMarker.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: None
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: REQ_EP_150

            Name:   Comment Marker Placement
            Desc:   CommentMarkers shall be placed only in EventMobSlots.  DescriptiveMarkers may be placed in any kind of MobSlot (e.g. TimelineMobSlot, EventMobSlot, StaticMobSlot).
            Node:   45 (CommentMarker)
            Result: Failed
            Reason: Mob slot with ID = 1 of Top-Level Composition "CommentMarker in Static Mob Slot" is not an Event Mob Slot but contains a CommentMarker.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: None
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: REQ_EP_150

            Name:   Documenting TaggedValues
            Desc:   Exporting applications shall document all TaggedValues in use using TaggedValueDefinition objects in the Dictionary::TaggedValueDefinitions property.
            Node:   6 (TaggedValue)
            Result: Failed
            Reason: TaggedValue "Test Case 1" is not documented in the dictionary.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: None
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: REQ_EP_151

            Name:   Documenting TaggedValues
            Desc:   Exporting applications shall document all TaggedValues in use using TaggedValueDefinition objects in the Dictionary::TaggedValueDefinitions property.
            Node:   14 (TaggedValue)
            Result: Failed
            Reason: TaggedValue "Test Case 2" is not documented in the dictionary.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: None
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: REQ_EP_151

            Name:   Documenting KLVData
            Desc:   Exporting applications shall document all KLVData in use in KLVDataDefinition objects in the Dictionary::KLVDefinitions property.
            Node:   66 (KLVData)
            Result: Failed
            Reason: KLVData with key 00000012-0012-0012-1212121212121212 is not documented in the dictionary.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: None
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: REQ_EP_152

